shared/util/arrays.js

```diff
@@ -6,13 +6,13 @@ export function intersperseFn<A, B>(
   arr: Array<A>
 ): Array<A | B> {
   if (arr.length === 0) {
-    return []
+    // $FlowIssue an array with no length is a valid type for any Array
+    return arr
   }
 
-  return arr.slice(1).reduce(
-    (acc, x, i, a) => {
-      return acc.concat([separatorFn(i, x, a), x])
-    },
-    [arr[0]]
-  )
+  let toReturn = [arr[0]]
```

Comment:
Not that I think we lose a lot of perf here ever but likely faster
toReturn = new Array(arr. length)
For...
toReturn[i] = ...

Reply:
I forgot you could size hint the array!